flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/data/DataStructureConvertersTest.java

```diff
@@ -205,20 +207,25 @@
 			TestSpec
 				.forDataType(ARRAY(BOOLEAN()))
 				.convertedTo(Boolean[].class, new Boolean[]{true, null, true, true})
+				.convertedTo(List.class, Arrays.asList(true, null, true, true))
 				.convertedTo(ArrayData.class, new GenericArrayData(new Boolean[]{true, null, true, true})),
 
 			TestSpec
-				.forDataType(ARRAY(INT().notNull()))
+				.forDataType(ARRAY(INT().notNull().bridgedTo(int.class))) // int.class should not have an impact
 				.convertedTo(int[].class, new int[]{1, 2, 3, 4})
-				.convertedTo(Integer[].class, new Integer[]{1, 2, 3, 4}),
+				.convertedTo(Integer[].class, new Integer[]{1, 2, 3, 4})
+				.convertedTo(List.class, new LinkedList<>(Arrays.asList(1, 2, 3, 4))),
 
 			// arrays of TINYINT, SMALLINT, INTEGER, BIGINT, FLOAT, DOUBLE are skipped for simplicity
 
 			TestSpec
 				.forDataType(ARRAY(DATE()))
 				.convertedTo(
 					LocalDate[].class,
-					new LocalDate[]{null, LocalDate.parse("2010-11-12"), null, LocalDate.parse("2010-11-12")}),
+					new LocalDate[]{null, LocalDate.parse("2010-11-12"), null, LocalDate.parse("2010-11-12")})
+				.convertedTo(
+					List.class,
+					new LinkedList<>(Arrays.asList(null, LocalDate.parse("2010-11-12"), null, LocalDate.parse("2010-11-12")))),
```

Comment:
Why do we need `LinkedList` here?

Reply:
It revealed another bug around `List.toObject`. `ArrayList.toObject` is always `internalArray.clone()` whereas LinkedList is always `Object[]`.